docs/modular-architecture.md

```diff
@@ -3,23 +3,40 @@ Below is a high-level description of the modular architecture of Fedimint.
 
 ## Dependency Tree
 ```mermaid
-graph BT;
-    fedimint/client-->mapi;
-    fedimint/server-->mapi;
-    fedimint/client-->mclient;
-    fedimint/server-->mserver;
-    mapi[[module/api]]-->core/api;
-    mclient[[module/client]]-->core/client;
-    mserver[[module/server]]-->core/server;
-    core/client-->core/api;
-    core/server-->core/api;
+flowchart BT;
+    fedimintd[[fedimintd: Final server binary]]
+    fedimint-server(fedimint-server: Fedimint server - abstract over modules)
+    fedimint-app[[fedimint-app: Final client application]]
+    fedimint-core(fedimint-core: Traits & glue between modules and generic code)
+    fedimint-api(fedimint-api: API defintion between Fedimint client & server)
+    fedimint-lib(fedimint-lib: Client side library - abstact over modules)
+
+    module-api("&lt;module>-api: Shared Fedimint module code & data")
+    module-client("&lt;module>-client: Client side Fedimint module")
+    module-server("&lt;module>-server: Server side Fedimint module")
+    
+    module-server-->module-api
+    module-api-->fedimint-core
+
+    fedimint-server-->fedimint-api
+    fedimint-server-->fedimint-core
+    fedimint-core-->fedimint-api
+    fedimintd-->module-server
+    fedimintd-->fedimint-server
+
+    module-client-->module-api
+    fedimint-lib-->fedimint-core
+    fedimint-lib-->fedimint-api
+    fedimint-app-->module-client
+    fedimint-app-->fedimint-lib  
+    fedimint-app-->fedimint-core
 ```
 
 ## Crate Descriptions
 `core` are the libraries that we make available to module developers
-- `core/api` - module type interfaces (txs, outcomes, consensus items, etc.), database, and API encoding/decoding
-- `core/client` - client interfaces and logic (tx builder, query strategies)
-- `core/server` - server interfaces and logic (federation module, consensus/hbbft code)
+- `core/api` - module type interfaces (txs, inputs, output, etc.), database, and API encoding/decoding
+- `core/client` - client interfaces, types, and logic (tx builder, query strategies)
+- `core/server` - server interfaces, types, and logic (federation module, outcomes, consensus items, consensus/hbbft code)

```

Comment:
This section seems out of date with the diagram

Reply:
Yes, I have modified the graph to move basic descriptions into the graph itself and I didn't update the lists.